mlflow/models/model.py

```diff
@@ -715,8 +715,19 @@ def log(
                             )
                     except Exception as e:
                         _logger.warning("Failed to load model config from %s: %s", model_config, e)
+
+                try:
+                    from mlflow.models.utils import _flatten_nested_params
+
+                    # We are using the `/` separator to flatten the nested params
+                    # since we are using the same separator to log nested metrics.
+                    flattened_params = _flatten_nested_params(model_config, sep="/")
+                except Exception as e:
+                    _logger.warning("Failed to flatten nested params: %s", str(e))
+                    flattened_params = model_config
```

Comment:
i think this variable name is not accurate. in the else case, the params are not flattened.